### ISSUE TITLE
Crash in version 1.0.0

### ISSUE THREAD

**Issue by brijohn** [GryBsh/NabuNetworkEmulator#10]
I was trying your recent 1.0.0 release the other day and after my network apparently suffered a small hiccup I noticed that the instance of the server I had running had crashed with the following stack trace. It seems that it tried to connect to could.nabu.ca and when it couldn't the whole program crashed with an unhandled exception.

```
Nov 25 23:25:15 bananapim2zero nns[20264]: Fail: [2023-11-25 23:25:15.4560] NabuNetwork:
Nov 25 23:25:18 bananapim2zero nns[20264]: Unhandled exception. System.Net.Http.HttpRequestException: No route to host (cloud.nabu.ca:443)
Nov 25 23:25:18 bananapim2zero nns[20264]:  ---> System.Net.Sockets.SocketException (113): No route to host
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|281_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cance>
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellation>
Nov 25 23:25:18 bananapim2zero nns[20264]:    --- End of inner exception stack trace ---
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellation>
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(QueueItem queueItem)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Http.HttpConnectionPool.HttpConnectionWaiter`1.WaitForConnectionAsync(Boolean async, CancellationToken requestCancellationToken)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cance>
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.<SendAsync>g__Core|5_0(HttpRequestMessage request, CancellationToken cancellationToken)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.<SendAsync>g__Core|5_0(HttpRequestMessage request, CancellationToken cancellationToken)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Net.Http.HttpClient.GetStringAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at Nabu.Network.HttpCache.GetString(String uri)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at Nabu.Network.NabuNetwork.IsNabuCaList(String source, String uri)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at Nabu.Network.NabuNetwork.IsKnownListType(String name, String path)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at Nabu.Network.NabuNetwork.RefreshSources(RefreshType refresh)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Threading.ThreadPoolWorkQueue.Dispatch()
Nov 25 23:25:18 bananapim2zero nns[20264]:    at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
```

**Comment by GryBsh** [comment]
I'll have a fix out for this in the next few days.
I hit this issue shortly before you did; it's not your network, it's his site. 
It has dropouts, timeouts and certificate glitches with a surprising frequency.

**Comment by GryBsh** [comment]
This is fixed in release v1.1. You should now see a message like this:
Warn: [2023-12-16 17:12:07.5424] NabuNetwork: Caching failed, please try again later
